### ISSUE TITLE
Revision data - branch name doesn't matched

### ISSUE THREAD

**Issue by QuAzI** [gitextensions/gitextensions#9737]
### Environment

- Git Extensions 3.5.4.12724
- Build 65f01f39982ccb121c4df7d4b00b506939553958
- Git 2.33.0.windows.2
- Microsoft Windows NT 10.0.19044.0
- .NET Framework 4.8.4400.0
- DPI 120dpi (125% scaling)

### Issue description

Pattern matched only if present in commit message. E.g. only in merges with branch. But not from `Local branch name` or `Remote branch name`

### Steps to reproduce

Search in: [+] Message  [+] Local branch name  [+] Remote branch name
Search pattern: `(?<=(feature|dev|hotfix)\/)[-\w]+`
Nested pattern: `[-\w]+`

Then create branch with template: `feature/12345-issue-name`

![image](https://user-images.githubusercontent.com/1897605/142394669-1f0a2a8a-c736-4f93-a782-5eac6f42921f.png)

**Comment by mstv** [comment]
Yes, that's how it works. The branch name is attached to its head commit only. Git does not store the destination branch of commits. There is a heuristic in the graph tooltips - far from being perfect. 
But a related additional git command is run in order to fill the section `Contained in branches` of the CommitInfo panel.
Though that list can be very long and will yield misleading matches.

**Comment by gerhardol** [comment]
Adding a note to documentation:
https://git-extensions-documentation.readthedocs.io/en/main/settings.html?highlight=shell#git-extensions-revision-data

`Note that the branch name is only searchable in the branch heads.`
